### PR TITLE
Release notes for 3.1.6/3.1.4 and Wireless Upload tutorial

### PR DESCRIPTION
**Pull request by edjubuh**
Opening this PR much earlier than usual since the website update is more interesting this time.

### CODE REVIEW

File: v5/releases/index.rst [hunk 1]
```diff
@@ -6,7 +6,6 @@ Kernel Releases
 ===============
 
 .. postlist:: 5
-   :sort:
```
Comment (edjubuh) [inline v5/releases/index.rst]: We will finally have 6 releases. An apparently `:sort:` makes it chronological from
oldest
...
newest

when we want the other way around

File: v5/releases/cli3.1.4.rst [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+-----
+
+See below for the new recommended method of installing the CLI
+
+macOS installs
+~~~~~~~~~~~~~~
+
+Install Homebrew if it's not already installed. This may take a while.
+Run brew tap purduesigbots/pros
+Run brew cask install gcc-arm-embedded pros-editor (if you want to install only the CLI, you can run brew cask install gcc-arm-embedded && brew install pros-cli instead)
```
Comment (HotelCalifornia) [inline v5/releases/cli3.1.4.rst]: need to update this. Homebrew Cask recently removed gcc-arm-embedded, so my recommended way to install the toolchain for now is to run `brew tap osx-cross/arm && brew install gcc-arm-bin` (see [relevant docs page](https://pros.cs.purdue.edu/v5/getting-started/macos.html))

File: v5/releases/cli3.1.4.rst [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+Notes:
+~~~~~~
+
+- if Homebrew complains about some files existing when trying to install the toolchain (gcc-arm-embedded), run rm -f /usr/local/bin/arm-none-eabi-* to clear the old files out
```
Comment (HotelCalifornia) [inline v5/releases/cli3.1.4.rst]: I think it's safe to remove this line now. This was intended for people migrating from the old 2.x mac installer where I had set it to install the toolchain here

File: v5/releases/index.rst (hunk 1 above)
Comment (HotelCalifornia) [reply v5/releases/index.rst]: will it autosort them correctly then?

File: v5/releases/kernel3.1.6.rst [hunk 4]
```diff
@@ -0,0 +1,40 @@
+=========================
+PROS Kernel 3.1.6 Release
+=========================
+
+.. post:: 12 February, 2019
+   :tags: blog, kernel-release
+
+**Be sure to look at important upgrade instructions below.**
```
Comment (HotelCalifornia) [inline v5/releases/kernel3.1.6.rst]: maybe make this one of the actual callout things (like note or warning)

File: v5/tutorials/topical/wireless-upload.rst [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    IS_LIBRARY:=0
+    # TODO: CHANGE THIS!
+    LIBNAME:=libbest
+    VERSION:=1.0.0
+    # EXCLUDE_SRC_FROM_LIB= $(SRCDIR)/unpublishedfile.c
+
+Change `IS_LIBRARY` to ``1`` and rename the library to something of your choosing. We recommend ``lib<robot name>``
+or ``lib<team number>`` (e.g. ``lib7701``) to be unique enough to avoid cause naming conflicts with other cold 
+images. By default, the Makefile is set up to exclude your project's opcontrol.cpp, autonomous.cpp, and 
+initialize.cpp files. If you change other files frequently, you can add lines like
```
Comment (HotelCalifornia) [inline v5/tutorials/topical/wireless-upload.rst]: > If you chance other files frequently

This is unclear to me. What does changing files frequently have to do with excluding them from the library?

File: v5/releases/cli3.1.4.rst (hunk 3 above)
Comment (edjubuh) [reply v5/releases/cli3.1.4.rst]: Will you submit suggestions for what Mac should look like now?

Comment (HotelCalifornia) [reply v5/releases/cli3.1.4.rst]:
```suggestion
```

File: v5/releases/cli3.1.4.rst [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+macOS
+-----
+
+See below for the new recommended method of installing the CLI
+
+macOS installs
+~~~~~~~~~~~~~~
+
+Install Homebrew if it's not already installed. This may take a while.
+Run brew tap purduesigbots/pros
```
Comment (HotelCalifornia) [inline v5/releases/cli3.1.4.rst]:
```suggestion
Run ``brew tap osx-cross/arm && brew tap purduesigbots/pros``
```

File: v5/releases/cli3.1.4.rst (hunk 2 above)
Comment (HotelCalifornia) [reply v5/releases/cli3.1.4.rst]:
```suggestion
Run ``brew install gcc-arm-bin && brew cask install pros-editor`` (if you want to install only the CLI, you can run ``brew install gcc-arm-bin pros-cli`` instead)

File: v5/tutorials/topical/wireless-upload.rst [hunk 7]
```diff
@@ -0,0 +1,65 @@
+====================================
+Wireless Upload and Hot/Cold Linking
+====================================
+
+.. note:: Wireless Upload requires at least CLI 3.1.4, Kernel 3.1.6, VEXos 1.0.5, and OkapiLib 3.1.7 (if used).
```
Comment (Octogonapus) [inline v5/tutorials/topical/wireless-upload.rst]: 3.1.7 isn't an OkapiLib version, which version are you referring to?

File: v5/tutorials/topical/wireless-upload.rst [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+    VERSION:=1.0.0
+    # EXCLUDE_SRC_FROM_LIB= $(SRCDIR)/unpublishedfile.c
+
+Change `IS_LIBRARY` to ``1`` and rename the library to something of your choosing. We recommend ``lib<robot name>``
+or ``lib<team number>`` (e.g. ``lib7701``) to be unique enough to avoid cause naming conflicts with other cold 
+images. When compiling, PROS will include this library as part of the cold image. Your library should only contain
+files you infrequently change so that you do not have to frequently upload the cold image. By default, the Makefile 
+is set up to exclude your project's opcontrol.cpp, autonomous.cpp, and initialize.cpp files. If you change other 
+files frequently, you can add lines like ``EXCLUDE_SRC_FROM_LIB+=$(SRCDIR)/myfile.c`` as needed.
+**An important caveat** is that code that goes into the cold image is **not** able to link against anything (call
```
Comment (Octogonapus) [inline v5/tutorials/topical/wireless-upload.rst]: What happens when I forget about this and try building? Do I get a linker error? Are there runtime problems?

File: v5/tutorials/topical/wireless-upload.rst (hunk 7 above)
Comment (HotelCalifornia) [reply v5/tutorials/topical/wireless-upload.rst]: whatever version you want us to- something that supports the hot/cold linking though preferable

Comment (Octogonapus) [reply v5/tutorials/topical/wireless-upload.rst]: I am inclined to say v4.0.0. Which may require that I bump odometry to v4.1.0, though that is something I am willing to do to get this stuff out sooner.

File: v5/tutorials/topical/wireless-upload.rst (hunk 8 above)
Comment (edjubuh) [reply v5/tutorials/topical/wireless-upload.rst]: Linker error. Is it ok to just leave it at
> Doing so will result in a linker error.
Or should I give a whole explanation? They'll see an error message along the lines of "myfunction: could not find symbol opcontrol" (I don't remember the exact verbage)

File: v5/tutorials/topical/wireless-upload.rst (hunk 7 above)
Comment (edjubuh) [reply v5/tutorials/topical/wireless-upload.rst]: OK you got time. We can finalize these versions before actually publishing

File: v5/tutorials/topical/wireless-upload.rst (hunk 8 above)
Comment (Octogonapus) [reply v5/tutorials/topical/wireless-upload.rst]: I think saying you will get a linker error and showing an example of such an error would be sufficient.

File: v5/tutorials/topical/wireless-upload.rst [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+Generally, if the program appears to be running correctly (i.e. the screen shows), then the compilation mode worked 
+correctly and the error you're experiencing is likely related to your code's logic. If your code's logic runs correctly
+when not using hot/cold linking, contact us so we may assist in troubleshooting.
+
+If you see a black scren, then PROS did not boot correctly.
+
+- A global constructor is in an infinite loop or raised an exception.
+- See also troubleshooting steps below
+
+If you see a grey screen, the PROS booted correctly, but is not running your hot image.
```
Comment (Octogonapus) [inline v5/tutorials/topical/wireless-upload.rst]:
```suggestion
If you see a grey screen, then PROS booted correctly, but is not running your hot image.
```

File: v5/tutorials/walkthrough/uploading.rst [hunk 10]
```diff
@@ -1,5 +0,0 @@
-=========
```
Comment (HotelCalifornia) [inline v5/tutorials/walkthrough/uploading.rst]: are we waiting for interactive UI to bring this back?

Comment (edjubuh) [reply v5/tutorials/walkthrough/uploading.rst]: I dunno. I don't know what to put in here. You click da upload button in the menu or on the toolbar.

Comment (HotelCalifornia) [reply v5/tutorials/walkthrough/uploading.rst]: I don't disagree, but there have been a surprising (to me) number of people looking for "the uploading docs."

at one point we had discussed adding some guide with screenshots of the editor (possibly that just meant updating the clawbot guide)

File: v5/releases/cli3.1.4.rst [hunk 11]
```diff
@@ -5,12 +5,12 @@ PROS CLI 3.1.4 Release
 .. post:: 12 February, 2019
    :tags: blog, cli-release
 
-- Added support for supplying upload arguments to ``mu``/``mut``/``ut``
-- Add V5 screen capture with ``prosv5 v5 capture``
 - New UI experience for creating, upgrading, and uploading projects
 - Add binary compression to save space on the V5
 - Add support for hot/cold linking
+- Add V5 screen capture with ``prosv5 v5 capture``
```
Comment (HotelCalifornia) [inline v5/releases/cli3.1.4.rst]: was this not ever released before?

Comment (edjubuh) [reply v5/releases/cli3.1.4.rst]: No

Comment (HotelCalifornia) [reply v5/releases/cli3.1.4.rst]: huh.

File: v5/releases/cli3.1.4.rst [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+
+Windows
+-------
+
+Download and run the appropriate installer
+
+macOS
+-----
+
+See below for the new recommended method of installing the CLI
```
Comment (HotelCalifornia) [inline v5/releases/cli3.1.4.rst]: No real need to split this into two sections
```suggestion
To upgrade a previous installation, run ``brew upgrade pros-cli``

To install from scratch:
```
(would need to also delete line 34/35)

File: v5/releases/cli3.1.4.rst [hunk 13]
```diff
@@ -0,0 +1,46 @@
...
+Install Homebrew if it's not already installed. This may take a while.
+Run ``brew tap osx-cross/arm && brew tap purduesigbots/pros``
+Run ``brew install gcc-arm-bin && brew cask install pros-editor`` (if you want to install only the CLI, you can run ``brew install gcc-arm-bin pros-cli`` instead)
+
+Notes:
+~~~~~~
+
+- if you already have the PROS CLI installed, or if Homebrew complains about some files existing when trying to install the CLI, you'll want to uninstall it first
+- if you installed using pip, you can run pip3 uninstall pros-cli-v5
+- if you installed using the .app bundle, you can move that bundle to the trash folder, and then run rm -f /usr/local/bin/prosv5 to clear out the old file
```
Comment (HotelCalifornia) [inline v5/releases/cli3.1.4.rst]:
```suggestion
- if you installed using the .app bundle, you can move that bundle to the trash folder, and then run ``rm -f /usr/local/bin/prosv5`` to clear out the old file
```

File: v5/releases/cli3.1.4.rst [hunk 14]
```diff
@@ -29,7 +29,9 @@ Download and run the appropriate installer
 macOS
 -----
 
-See below for the new recommended method of installing the CLI
+To upgrade a previous installation, run ``brew upgrade pros-cli``
+
+To install from scratch:
 
 macOS installs
 ~~~~~~~~~~~~~~
```
Comment (HotelCalifornia) [inline v5/releases/cli3.1.4.rst]: still need to delete this line (35) and the one above it. I'd make a suggestion but i'm honestly not sure how to do multiline suggestions like that

File: v5/releases/kernel3.1.6.rst [hunk 15]
```diff
@@ -0,0 +1,40 @@
...
+
+Important Upgrade Instructions
+------------------------------
+
+In order to support modified hot/cold linking we needed to modify the Makefile, which
+is intended to be modified by you. When upgrading your project (regardless of whether
+you enable hot/cold linking), you must modify your Makefile as follows:
+
+1. Remove all lines after `-include ./common.mk`
+2. Add the following two lines above the `-include ./common.mk` line (perhaps below `EXTRA_CXXFLAGS=`):
```
Comment (HotelCalifornia) [inline v5/releases/kernel3.1.6.rst]:
```suggestion
2. Add the following two lines above the `-include ./common.mk` line (for example, below `EXTRA_CXXFLAGS=`):
```

File: v5/tutorials/topical/wireless-upload.rst (hunk 7 above)
Comment (Octogonapus) [reply v5/tutorials/topical/wireless-upload.rst]:
```suggestion
.. note:: Wireless Upload requires at least CLI 3.1.4, Kernel 3.1.6, VEXos 1.0.5, and OkapiLib 3.3.12 (if used).
```

**Pull request merged**